src/timer/timer_game_calendar.cpp

```diff
@@ -105,10 +105,13 @@ bool TimerManager<TimerGameCalendar>::Elapsed([[maybe_unused]] TimerGameCalendar
 
 	/* If we are using a non-default calendar progression speed, we need to check the sub_date_fract before updating date_fract. */
 	if (_settings_game.economy.minutes_per_calendar_year != CalendarTime::DEF_MINUTES_PER_YEAR) {
-		TimerGameCalendar::sub_date_fract++;
+		TimerGameCalendar::sub_date_fract += Ticks::DAY_TICKS;
```

Comment:
Do I read this right that `DAY_TICKS` is not really relevant here, but that it just defines the resolution of the sub-ticker? So it could also be, say, `1000`?

At least I have a hard time understanding what the relation with `DAY_TICKS` is here, but I might just not be understanding this :)

Reply:
The sizing of sub_date_fract and the limit of the _settings_game.economy.minutes_per_calendar_year implicitly assume that this is DAY_TICKS. (This is what the comment for MAX_MINUTES_PER_YEAR is talking about). The increment needs to be an integral multiple of DAY_TICKS for the algorithm to make sense (additional ticks per day are spread over DAY_TICKS date fracts), and 1 x DAY_TICKS is the smallest/least surprising value to use.

In general I'm not trying to justify the design of the wall-clock timekeeping mode, I'm merely aiming to fix this aspect to match what was (to me at least) clearly the intended behaviour.